File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        
        int[] answer = new int[2];
        
        int max = Math.max(n, m);
        int min = Math.min(n, m);
        
        answer[0] = gcd(max, min);
        answer[1] = max * min / answer[0];
        
        return answer;
    }
    
    static int gcd (int n, int m) {
        
        if (n % m == 0) return m;
        
        return gcd(m, n % m);
    }
}

File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def solution(n, m):
    # 유클리드 호제법: x와 y의 최대 공약수와 y와 x % y의 최대 공약수와 같다
    # 최소 공배수는 n * m를 최대 공약수로 나눈 몫이다
    x, y = max(n, m), min(n, m)
    i = 1
    while i > 0:
        i = x % y
        x, y = y, i
    return [x, int(n * m / x)]